### PR TITLE
fix: breaking changes are now recognized as meaningful

### LINKED ISSUE #139: 🚀 Feature: Return `true` for `BREAKING CHANGE`?
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/should-semantic-release/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

[Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) dictates that `BREAKING CHANGE` is meaningful, in that it denotes a X.0.0 version bump.

Should we add this to the list of always meaningful commits?

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by steveluscher**
Closes #139.

### CODE REVIEW

File: src/getCommitMeaning.ts [hunk 1]
```diff
@@ -8,22 +11,31 @@ const releaseCommitTester =
...
+			commit.notes.some(({ title }) => title === "BREAKING CHANGE")
+		) {
 			return "meaningful";
 		}
 
 		if (alwaysIgnoredTypes.has(type)) {
 			return { type };
 		}
+	} catch {
+		/* empty */
```
Comment (JoshuaKGoldberg) [inline src/getCommitMeaning.ts:32]: [Refactor] I'm not a big fan of `try`/`catch` as a control flow concept. It's not type-safe, is hard to mentally model, and can accidentally wrap around unintentionally throwing code too. Let's restrict what's being wrapped to just the `toConventionalChangelogFormat` call.

File: src/getCommitMeaning.ts [hunk 2]
```diff
@@ -1,4 +1,7 @@
-import conventionalCommitsParser from "conventional-commits-parser";
+import {
+	parser,
+	toConventionalChangelogFormat,
+} from "@conventional-commits/parser";
```
Comment (JoshuaKGoldberg) [inline src/getCommitMeaning.ts:4]: [Refactor] 🤔 `@conventional-commits/parser` hasn't had a commit or sustained GitHub issue triage in 3 years, and uses `throw` as a form of control flow. Whereas `conventional-commits-parser` released 3 months ago. I'm a little averse to going down to an older, less-maintained package if it's avoidable.

What do you think about instead using a small regex to check for a subject, optionally an area/scope, and `!` at the beginning? `/^\w+(\(\w+\))?!:/` or something like that?

File: src/getCommitMeaning.test.ts [hunk 3]
```diff
@@ -40,6 +40,17 @@ describe("getCommitMeaning", () => {
...
+		["chore!: message", "meaningful"],
+		["docs!: message", "meaningful"],
+		["style!: message", "meaningful"],
+		["chore!: release", "meaningful"],
+		["chore: bump\n\nBREAKING CHANGE: breaks things", "meaningful"],
+		["chore: bump\n\nBREAKING-CHANGE: breaks things", "meaningful"],
+		["docs: bump\n\nBREAKING CHANGE: breaks things", "meaningful"],
+		["docs: bump\n\nBREAKING-CHANGE: breaks things", "meaningful"],
+		["style: bump\n\nBREAKING CHANGE: breaks things", "meaningful"],
+		["style: bump\n\nBREAKING-CHANGE: breaks things", "meaningful"],
```
Comment (JoshuaKGoldberg) [inline src/getCommitMeaning.test.ts:53]: [Testing] Thinking about test edge cases...

* `!` in the commit body (so not a breaking/meaningful change)
* Footer note other than `BREAKING CHANGE`
* Multiple footer notes

File: src/getCommitMeaning.test.ts [hunk 4]
```diff
@@ -40,6 +40,17 @@ describe("getCommitMeaning", () => {
 		["chore(deps): release", "release"],
 		["chore(deps): release 1.2.3", "release"],
 		["chore(deps): release v1.2.3", "release"],
+		// Breaking changes are meaningful no matter the type of the commit.
```
Comment (JoshuaKGoldberg) [inline src/getCommitMeaning.test.ts:43]: [Style] Just nitpicking here, I think the comment is explanatory from the test cases.

```suggestion
```

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
🚀 A good start, thanks for getting it going! Left a few points of discussion - what do you think?

**Comment by steveluscher** [comment]
This is all great feedback! Thanks for taking the time to go over it.

I'm going to step away from this PR though instead of finishing it up.

giphy
